src/main/java/com/example/algorithm/Algorithm.java

```diff
@@ -12,4 +12,5 @@ public interface Algorithm {
     StepReport step();
     boolean isFinished();
     BooleanProperty getIsFinishedProperty();
+    void checkIsFinished();
```

Comment:
I am not a big fan of this function, because it requires explicit call to change algorithm state. Is it necessary?

Reply:
I'm not a fan either but there are bindings between `isFinished` property and buttons so if we set it during algorithm's execution, those buttons will be active before animation is over.

Reply:
We can also get rid of bindings and change it manually in simulation.

Reply:
Added idle dependecy for start button. So this function isn't neccessary any more?